### PR TITLE
Expose XSC git info API

### PR DESCRIPTION
**Pull request by orz25**
- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Remove SendScanGitInfoContext function call to cli-security package in order to have one multi scan id to multiple sca scans (we send a scan request per technology and we wish to associate all the scans to one multi scan id).

PRs for cli-security and frogbot:
cli-security: https://github.com/jfrog/jfrog-cli-security/pull/20
frogbot:  https://github.com/jfrog/frogbot/pull/633

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
Note that the *Static Analysis* check is failing.

File: xray/manager.go [hunk 1]
```diff
@@ -204,3 +204,9 @@ func (sm *XrayServicesManager) XscEnabled() (string, error) {
 	scanService.XrayDetails = sm.config.GetServiceDetails()
 	return scanService.IsXscEnabled()
 }
+
+func (sm *XrayServicesManager) SendXscGitInfoRequest(details *services.XscGitInfoContext) (multiScanId string, err error) {
```
Comment (attiasas) [inline xray/manager.go:208]: please add comment explaning this request

**Review by attiasas — CHANGES_REQUESTED** [review]
Nice, take a look at my comments.
Please change the PR name to fit the changes of exposing new API.
Also add to the README the new changes of how to use it.

File: tests/xrayscan_test.go [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
+	GitRepoUrl: "https://git.jfrog.info/projects/XSC/repos/xsc-service",
+	BranchName: "feature/XRAY-123-cool-feature",
+	CommitHash: "acc5e24e69a-d3c1-4022-62eb-69e4a1e5",
+}
+var gitInfoContextWithMissingFields = xrayServices.XscGitInfoContext{
+	GitRepoUrl: "https://git.jfrog.info/projects/XSC/repos/xsc-service",
+	BranchName: "feature/XRAY-123-cool-feature",
+}
+var testMultiScanId = "3472b4e2-bddc-11ee-a9c9-acde48001122"
+var testXscVersion = "1.0.0"
```
[the remark this answers is not in the document]
Comment (orz25) [reply tests/xrayscan_test.go:27]: Should I move also `gitInfoContextWithMinimalRequiredFields` and `gitInfoContextWithMissingFields`?

[the remark this answers is not in the document]
Comment (attiasas) [reply tests/xrayscan_test.go:27]: @orz25, Yes all of the constants for tests are there you can see. see similar tests at the same file

File: tests/xrayscan_test.go [hunk 3]
```diff
@@ -60,3 +64,69 @@ func scanBuild(t *testing.T, buildName, buildNumber, expected string) {
...
+func TestIsXscEnabled(t *testing.T) {
+	xrayServerPort, xrayDetails, client := initXrayScanTest(t)
+	testsScanService = xrayServices.NewScanService(client)
+	testsScanService.XrayDetails = xrayDetails
+	testsScanService.XrayDetails.SetUrl("http://localhost:" + strconv.Itoa(xrayServerPort) + "/xray/")
+
+	result, err := testsScanService.IsXscEnabled()
+	if err != nil {
+		t.Error(err)
+	}
```
Comment (yahavi) [inline tests/xrayscan_test.go:77]:
```suggestion
        assert.NoError(t, err)
```

File: tests/xrayscan_test.go [hunk 4]
```diff
@@ -60,3 +64,69 @@ func scanBuild(t *testing.T, buildName, buildNumber, expected string) {
...
+
+	if result != xray.TestXscVersion {
+		t.Error("Expected:", xray.TestXscVersion, "Got: ", result)
+	}
```
Comment (yahavi) [inline tests/xrayscan_test.go:81]: Let's use assert.Equals(t, ....)

File: tests/xrayscan_test.go [hunk 5]
```diff
@@ -60,3 +64,69 @@ func scanBuild(t *testing.T, buildName, buildNumber, expected string) {
...
+func sendGitInfoContext(t *testing.T, gitInfoContext *xrayServices.XscGitInfoContext, expected string) {
+	result, err := testsScanService.SendScanGitInfoContext(gitInfoContext)
+	if err != nil {
+		assert.ErrorContains(t, err, expected)
+		return
+	}
+
+	if result != expected {
+		t.Error("Expected:", expected, "Got: ", result)
+	}
```
Comment (yahavi) [inline tests/xrayscan_test.go:115]: Let's use assert.Equals(t, ....)

File: tests/xrayscan_test.go [hunk 6]
```diff
@@ -60,3 +64,69 @@ func scanBuild(t *testing.T, buildName, buildNumber, expected string) {
...
+	xrayServerPort = xray.StartXrayMockServer(t)
+	xrayDetails = GetXrayDetails()
+	client, err = jfroghttpclient.JfrogClientBuilder().
+		SetClientCertPath(xrayDetails.GetClientCertPath()).
+		SetClientCertKeyPath(xrayDetails.GetClientCertKeyPath()).
+		AppendPreRequestInterceptor(xrayDetails.RunPreRequestFunctions).
+		Build()
+	if err != nil {
+		t.Error(err)
+	}
```
Comment (yahavi) [inline tests/xrayscan_test.go:130]:
```suggestion
        assert.NoError(t, err)
```

**Review by yahavi — CHANGES_REQUESTED** [review]

**Pull request merged**